src/trigger.c

```diff
@@ -53,8 +54,8 @@ ts_trigger_create_on_chunk(Oid trigger_oid, char *chunk_schema_name, char *chunk
 	} while (0);
 
 	Assert(IsA(stmt, CreateTrigStmt));
-	stmt->relation->relname = chunk_table_name;
-	stmt->relation->schemaname = chunk_schema_name;
+	stmt->relation->relname = pstrdup(chunk_table_name);
+	stmt->relation->schemaname = pstrdup(chunk_schema_name);
```

Comment:
why do we need the copy now?

Reply:
PostgresSQL more-or-less always expects mutable objects/strings in planner nodes. The trigger function called here now takes non-mutable strings parameters (`chunk_table_name` and `chunk_schema_name`), so copies are needed to guarantee that the original strings are not modified.

IMO, this is a good thing as it is much safer and prevents a number of subtle bugs.

Reply:
IMO we should not unnecessarily copy objects just to preserve const correctness that is not enforced by PG APIs. This PR introduces multiple unnecessary pstrdups.  It is preferable to either a)  explicitly cast away const i.e. stmt->relation->relname = (char*)chunk_table_name if we want to use const char* arg types in our APIs. or b) Not introduce const char* where it cannot be passed down to the underlying PG function.

Reply:
FWIW, our code deviates from common PostgreSQL practice by overuse of references instead of safely copying strings and other objects. Here's just a few examples in PostgreSQL:

https://github.com/postgres/postgres/blob/caba8f0d43fb679c6f9643456080408a6bc370e8/src/backend/commands/tablecmds.c#L2618

https://github.com/postgres/postgres/blob/caba8f0d43fb679c6f9643456080408a6bc370e8/src/backend/commands/trigger.c#L1901

https://github.com/postgres/postgres/blob/caba8f0d43fb679c6f9643456080408a6bc370e8/src/backend/foreign/foreign.c#L142

pstrdup is used extensively in PostgreSQL itself, and is cheap given the memory context system. So, I am not sure what we are trying to optimize for by minimizing pstrdups. In fact, we freely heap-allocate and copy other objects in the code and people generally don't mind. Every makeXYZ command and palloc allocates memory in the same way, but for some reason we try hard to not allocate strings even though they are referenced in a palloced object.